### PR TITLE
Release/0.10.0

### PR DESCRIPTION
**Pull request by adatzer**
Notes:

Upgrading the yaml library from v0 means we don't any more support invalid yaml syntax, for example `:test: [x]`. Before, that test case would have been unmarshalled to `"[x]"`. Now it will fail. 

- Current yaml libraries versions do not support the old v0 behaviour, neither provide a way to do so. 
- For this PR, in order to not break backwards compatibility on that (still invalid) behaviour, the alternatives considered (custom Unmarshaler, regex replace) were flaky enough to reduce the decision to either upgrade or not. Since the yaml v0 has not been updated for some years and since the previous behaviour is invalid to the syntax/parser level, this PR includes an upgrade.
- Additional considerations: Unmarshalling to `map[string]interface{}` is variable. For example, `gopkg.in/yaml.v3` decodes unquoted `2022-01-01` to `time.Time`, which is another potentially breaking change (see data-models playbooks). For that reason, the library chosen to upgrade to was `github.com/goccy/go-yaml`, that was found to be the most friction-less for the transition.

Still todo:

- [x] Some more local testing with data models
- [x] Prepare for release commit
- [x] snyk test locally

### CODE REVIEW

File: integration/run_tests.sh [hunk 1]
```diff
@@ -131,8 +131,8 @@ assert_ExitCodeForCommand "6" "${bin_path} -playbook ${root_key}/good-postgres-w
 assert_ExitCodeForCommand "6" "${bin_path} -playbook ${root_key}/good-postgres-with-template.yml"
 assert_ExitCodeForCommand "0" "${bin_path} -playbook ${root_key}/good-postgres-with-template.yml -var username=postgres,password=,host=localhost"
 
-# Test: Truncated steps field in playbook should return exit code 8
-assert_ExitCodeForCommand "8" "${bin_path} -playbook ${root_key}/good-postgres-truncated.yml -lock ${root}/dist/integration-lock"
+# Test: Truncated steps field in playbook should return exit code 1
```
Comment (colmsnowplow) [inline integration/run_tests.sh:134]: Not that it's wrong but any particular reason to change from 8 to 1 here?

**Review by colmsnowplow — APPROVED** [review]
LGTM.

Two things I'd like to cover when we speak tomorrow - both are more general than about the PR.

- Rollout - how do we plan to do so & what's the urgency of doing so
- Logging & data - I thought this PR was about ensuring we don't log what we shouldn't - but I might've just been confused (or we might just already not be doing so). I'd love to get a refresher on that topic to keep myself up to speed.

File: integration/run_tests.sh (hunk 1 above)
Comment (adatzer) [reply integration/run_tests.sh:134]: The reason is that now invalid/truncated playbooks fail from [this line](https://github.com/snowplow/sql-runner/blob/8279f9654f315de25ab4b7c1598e583ee18a62bf/sql_runner/main.go#L58). The thought was to match the other exit scenarios from main (through `log.Fatal`) that happen before `Run`. Also, exit code `8` is tied to "no queries to run", which may not exactly reflect "no targets" or "no steps". Not a strong opinion on this though, happy to change!

**Comment by adatzer** [comment]
Thank you for reviewing @colmsnowplow !
As we discussed, the plan is to go for `0.10.0` now and return to these issues at a next release.

**Comment by adatzer** [comment]
@colmsnowplow , just re-requested your review, because of the last-minute sneaky commit to close https://github.com/snowplow/sql-runner/issues/210 .  Everything else is the same since you last reviewed, but for a `make format`.

**Review by colmsnowplow — APPROVED** [review]
LGTM! 🚢

**Pull request merged**